src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/RoleAssignmentFilteringResult.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import static uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType.CHALLENGED;
+import static uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType.SPECIFIC;
+import static uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType.STANDARD;
+
+@Data
+@AllArgsConstructor
+public class RoleAssignmentFilteringResult {
+    private final RoleAssignment roleAssignment;
+    private final RoleMatchingResult roleMatchingResult;
+
+    public AccessProcess getAccessProcess(String grantType) {
+        if (STANDARD.name().equals(grantType)
+        && SPECIFIC.name().equals(grantType)
+        && CHALLENGED.name().equals(grantType)
+        && roleMatchingResult.matchedAllValues()) {
+            return AccessProcess.NONE;
+        } else if (STANDARD.name().equals(grantType)
+            && roleMatchingResult.matchedAExceptRegionAndLocation()) {
+            return AccessProcess.CHALLENGED;
+        }
+        return AccessProcess.SPECIFIC;
+    }
```

Comment:
this is business logic, perhaps a Result object is not the right place for that. I'd say we remove it from now, and when we work on updating the endpoints that need to return that info we see where's the best place for this logic. Maybe a dedicated service

Reply:
Ok